client/containers/DashboardEdges/NewEdgeInput.tsx

```diff
@@ -47,59 +51,63 @@ const indeterminateMenuItem = (
+	const { usedPubIds, onSelectItem } = props;
 	const [queryValue, setQueryValue] = useState('');
-	const [suggestedItems, setSuggestedItems] = useState<any[]>([]);
 	const throttledQueryValue = useThrottled(queryValue, 250, true, true);
+	const [pubSearchTerm, setPubSearchTerm] = useState<null | string>(null);
+	const [proposedItem, setProposedItem] = useState<null | SuggestedItem>(null);
+
+	const {
+		allQueries: { isLoading },
+		currentQuery: { pubs },
+	} = useManyPubs({ query: { term: pubSearchTerm || '' }, batchSize: 5 });
+
+	const suggestedItems = proposedItem ? [proposedItem] : pubs.map((pub) => ({ targetPub: pub }));
 
 	useEffect(() => {
 		if (isUrl(throttledQueryValue) || isDoi(throttledQueryValue)) {
-			setSuggestedItems([{ indeterminate: true }]);
+			setProposedItem({ indeterminate: true });
 			apiFetch
-				// @ts-expect-error ts-migrate(2339) FIXME: Property 'get' does not exist on type '(path: any,... Remove this comment to see the full error message
 				.get(`/api/pubEdgeProposal?object=${encodeURIComponent(throttledQueryValue)}`)
 				.then((res) => {
 					if (res) {
-						setSuggestedItems([res]);
+						setProposedItem(res);
 					} else {
-						setSuggestedItems([{ createNewFromUrl: throttledQueryValue }]);
+						setProposedItem({ createNewFromUrl: throttledQueryValue });
 					}
 				});
-		} else if (throttledQueryValue) {
-			setSuggestedItems(
-				availablePubs
-					.filter(
-						(pub) =>
-							fuzzyMatchPub(pub, throttledQueryValue) && !usedPubIds.includes(pub.id),
-					)
-					.slice(0, 5)
-					.map((pub) => ({ targetPub: pub })),
-			);
 		} else {
-			setSuggestedItems([]);
+			setPubSearchTerm(throttledQueryValue);
+			setProposedItem(null);
 		}
-	}, [availablePubs, throttledQueryValue, usedPubIds]);
+	}, [throttledQueryValue, usedPubIds]);
 
-	const renderItem = (item, { handleClick, modifiers }) => {
-		const { externalPublication, targetPub, indeterminate } = item;
-		if (indeterminate) {
+	const renderItem = (item: SuggestedItem, { handleClick, modifiers }) => {
+		if ('indeterminate' in item && item.indeterminate) {
 			return indeterminateMenuItem;
 		}
-		if (targetPub) {
-			return (
-				<PubMenuItem
-					key={targetPub.title}
-					title={targetPub.title}
-					contributors={targetPub.attributions.filter((attr) => attr.isAuthor)}
-					image={targetPub.avatar}
-					active={modifiers.active}
-					onClick={handleClick}
-					showImage={true}
-				/>
-			);
+		if ('targetPub' in item) {
+			const { targetPub } = item;
+			if (targetPub) {
+				return (
+					<PubMenuItem
+						key={targetPub.id}
+						title={targetPub.title}
+						contributors={targetPub.attributions.filter((attr) => attr.isAuthor)}
+						image={targetPub.avatar}
+						active={modifiers.active}
+						onClick={handleClick}
+						showImage={true}
+					/>
+				);
+			}
 		}
-		if (externalPublication && externalPublication.title) {
-			const { title, contributors, avatar } = externalPublication;
+		if (
```

Comment:
Is this condition mutually exclusive with the targetPub condition above? In that case, `else if`?
Given each if clause contains a return statement, they seem mutually exclusive, `else` makes the logic easier to track.

Reply:
I agree that this would potentially add clarity, but our linter actually has [a rule](https://eslint.org/docs/rules/no-else-return) that breaks `if/else if/else` clauses into disjoint `if` statements if every branch has a `return` statement and it can statically verify that they're mutually exclusive. It's a style that I've gotten used to, but if it bothers you, you have the go ahead to turn it off.